### PR TITLE
Add a FAQ item about why Godot aims to keep its core feature set small

### PR DESCRIPTION
**Pull request by Calinou**
This is based on yesterday's conversation in the Godot contributors chat.

### CODE REVIEW

**Comment by akien-mga** [comment]
Binary size is one thing but it's not the main reason why we have to be selective with what we include. The problem is not with Godot becoming 200 MB for the editor, it's that all added code requires maintenance, increases compilation time, increases the potential surface area for bugs to be found in the engine, etc. More API to be familiar with to be able to contribute to a given area of the engine, etc.

I think that reducing it to file size concerns for download speeds is not describing the whole picture, even if it's one relevant factor.

**Comment by Calinou** [comment]
> Binary size is one thing but it's not the main reason why we have to be selective with what we include. The problem is not with Godot becoming 200 MB for the editor, it's that all added code requires maintenance, increases compilation time, increases the potential surface area for bugs to be found in the engine, etc. More API to be familiar with to be able to contribute to a given area of the engine, etc.
> 
> I think that reducing it to file size concerns for download speeds is not describing the whole picture, even if it's one relevant factor.

Indeed, I think it makes sense to describe the full picture more. I reworded the FAQ item entirely.

File: about/faq.rst [hunk 1]
```diff
@@ -207,6 +207,48 @@ In the long term, we may develop a Direct3D 12 renderer for Godot (mainly for
 the Xbox's purposes), but Vulkan and OpenGL will remain the default rendering
 backends on all platforms, including Windows.
 
+Why does Godot aim to keep its core feature set small?
+------------------------------------------------------
+
+Godot intentionally does not include features that can be implemented by add-ons
+unless they are used very often. A few examples of this would be artificial
+intelligence or terrain systems.
```
Comment (clayjohn) [inline about/faq.rst:215]: I wouldn't mention terrain here. The thinking around terrain has shifted and I think the only thing stopping it from being added is the amount of work necessary to do it properly/

**Review by clayjohn — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.

**Pull request merged**